driver-core/src/main/com/mongodb/client/model/MongoTimeUnit.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model;
+
+/**
+ * Units for specifying time-based bounds for {@linkplain Window windows} and output units for some time-based
+ * {@linkplain WindowedComputation windowed computations}.
+ *
+ * @since 4.3
+ */
+public enum MongoTimeUnit {
```

Comment:
`Unit` clashes with `scala.Unit` and is also not descriptive enough, `TimeUnit` clashes with `java.util.concurrent.Unit`. So I settled on `MongoTimeUnit`.

I also considered using `java.time.temporal.ChronoUnit` instead, but the downsides are by far greater than the small upside of reusing an existing class:

1. We would not support all its instances, and would have to document the subset that is supported. This is weird and not user-friendly (the `java.time` API does this, but there it is at least can be explained by the complexity and extensibility of the API).
2. This approach is incompatible with server starting to support time units that do not have `ChronoUnit` counterparts. It is possible to work around this limitation of `ChronoUnit` by using the `TemporalUnit` interface instead, but such an approach gives us no benefits compared to introducing a new `Enum` class.

Reply:
I'm happy with introducing our own class, but should we limit it to the window use-case by using the less generic "Window" prefix on the name instead of "Mongo"?  So it could be "WindowTimeUnit" or "WindowUnit"

Reply:
I thought about it, but decided to not go with that name for the following reasons:

1. These values are also used by the derivative/integral window functions when the variable is of BSON Date type, i.e., they are used not only for defining windows.
2. The [document specifying the syntax](https://docs.google.com/document/d/1IDbgaoEN9ttB9DNPk5mbRIT6S935QaL0HKlKGwOcSHE/edit#) specifies that BSON Date bounds for range windows are computed as per the [`$dateAdd`](https://docs.mongodb.com/v5.0/reference/operator/aggregation/dateAdd/) aggregation pipeline operation. This means that if we add builders for `$dateAdd`/`$dateDiff` operations we should use the same enum there, and having  it named `WindowTimeUnit` would be confusing. Nicholas Zolnierz (the author of the aforementioned document) confirmed that windows bounds should allow the same units allowed by `$dateAdd`.

P.S. I have just noticed that `$dateAdd` allows `quarter` as a unit, but range windows do not explicitly specify this unit among allowed units. The server, nonetheless, allows `quarter` as a unit for windows, and Nicholas Zolnierz confirmed that `quarter` is supposed to be allowed. I will add it to the enum.